inc/class_captcha.php

```diff
@@ -307,6 +312,40 @@ function validate_captcha()
 				}
 			}
 		}
+		elseif(in_array($this->type, array(8)))
+		{
+			$response = $mybb->input['g-recaptcha-response'];
+			if(!$response || strlen($response) == 0)
+			{
+				$this->set_error($lang->invalid_nocaptcha);
+			}
+			else
+			{
+				// We have a reCAPTCHA invisible to handle
+				// Contact Google and see if our reCAPTCHA was successful
+				$response = fetch_remote_file($this->verify_server, array(
+					'secret' => $mybb->settings['recaptchaprivatekey'],
+					'score' => $mybb->settings['recaptchascore'],
+					'remoteip' => $session->ipaddress,
+					'response' => $response
+				));
+
+				if($response == false)
```

Comment:
We'd normally prefer strict equality comparison in this case.

```suggestion
				if($response === false)
```